presto-hive/src/test/java/io/prestosql/plugin/hive/orc/TestOrcAcidPageSource.java

```diff
@@ -98,29 +99,45 @@ public void testFullFileSkipped()
     @Test
     public void testSomeStripesAndRowGroupRead()
     {
-        assertRead(ImmutableSet.copyOf(NationColumn.values()), OptionalInt.of(5));
+        assertRead(ImmutableSet.copyOf(NationColumn.values()), OptionalInt.of(5), Optional.empty(), nationKey -> false);
     }
 
-    private static void assertRead(Set<NationColumn> columns, OptionalInt nationKeyPredicate)
+    @Test
+    public void testDeletedRows()
+    {
+        Path partitionLocation = new Path(getClass().getClassLoader().getResource("nation_delete_deltas") + "/");
+        Optional<DeleteDeltaLocations> deltaDeltaLocations = DeleteDeltaLocations.builder(partitionLocation)
+                .addDeleteDelta(new Path(partitionLocation, deleteDeltaSubdir(3L, 3L, 0)), 3L, 3L, 0)
+                .addDeleteDelta(new Path(partitionLocation, deleteDeltaSubdir(4L, 4L, 0)), 4L, 4L, 0)
+                .build();
+
+        assertRead(ImmutableSet.copyOf(NationColumn.values()), OptionalInt.empty(), deltaDeltaLocations, nationKey -> nationKey == 5 || nationKey == 19);
+    }
+
+    private static void assertRead(Set<NationColumn> columns, OptionalInt nationKeyPredicate, Optional<DeleteDeltaLocations> deleteDeltaLocations, IntPredicate deletedRows)
     {
         TupleDomain<HiveColumnHandle> tupleDomain = TupleDomain.all();
         if (nationKeyPredicate.isPresent()) {
             tupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(toHiveColumnHandle(NATION_KEY), Domain.singleValue(INTEGER, (long) nationKeyPredicate.getAsInt())));
         }
 
-        List<Nation> actual = readFile(columns, tupleDomain);
+        List<Nation> actual = readFile(columns, tupleDomain, deleteDeltaLocations);
 
         List<Nation> expected = new ArrayList<>();
         for (Nation nation : ImmutableList.copyOf(new NationGenerator().iterator())) {
-            if (!nationKeyPredicate.isPresent() || nationKeyPredicate.getAsInt() == nation.getNationKey()) {
-                expected.addAll(nCopies(1000, nation));
+            if (nationKeyPredicate.isPresent() && nationKeyPredicate.getAsInt() != nation.getNationKey()) {
+                continue;
+            }
+            if (deletedRows.test((int) nation.getNationKey())) {
+                continue;
             }
+            expected.addAll(nCopies(1000, nation));
```

Comment:
Why? Looks like this belongs to `Add support for reading ORC ACID files` commit?

Reply:
Why nCopies? Each row of nation is replicated 1000 times in test file.

This is an extension to changes in `Add support for reading ORC ACID files ` commit, ones added here are not needed in that commit hence separate.